### PR TITLE
(fix) Workshop Assistants layout fix

### LINKED ISSUE #164: Layout issue on Josh’s workshop page
The assistance names at the bottom of Josh’s workshop page are running into each other and the pictures aren’t spaced evenly on iPhone 13 Pro Max.

### PR DESCRIPTION
**Pull request by mazenchami**
# Description

[Trello Card](179-layout-issue-on-joshs-workshop-page) — #164 

updating how we lay out all our assistants on the workshop screen

# Graphics

| Before |
|--------|
|  |

| iOS            | Android            |
| -------------- | ------------------ |
| <br/> | <br/> |

# Checklist:

- [x] I have done a thorough self-review of my code
- [ ] I have tested with a screen reader and font-scaling turned on and added necessary accessibility features
- [x] I have run tests and linter

### CODE REVIEW

File: app/screens/TalkDetailsScreen/AssistantsList.tsx [hunk 1]
```diff
@@ -36,7 +39,7 @@ export function AssistantsList(props: AssistantsListProp) {
                   socialButtons={[
                     { icon: "twitter", url: assistant.twitter },
                     { icon: "github", url: assistant.github },
-                    { icon: "link", url: assistant.externalURL },
+                    { icon: "link", url: assistant.externalURL || assistant.website },
```
Comment (mazenchami) [inline app/screens/TalkDetailsScreen/AssistantsList.tsx:42]: this was added in webflow and we're handling it elsewhere also

**Comment by morganick** [comment]
For the icons under the person's picture, we should center them otherwise they look slightly off center:

Screenshot_20230425_082634_Chain React App 2023

Maybe we should stack the assistants vertically instead so that we would have more space for their names?

Screenshot_20230425_082737_Chain React App 2023

If the assistant has all three social icons, they crowd each other on smaller devices. See also [pixelated icons issue](https://github.com/infinitered/ChainReactApp2023/issues/145)

@justinhuskey @JenFucci thoughts?

File: app/screens/TalkDetailsScreen/AssistantsList.tsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
-import { ImageStyle, TextStyle, View, ViewStyle } from "react-native"
+import { Dimensions, ImageStyle, TextStyle, View, ViewStyle } from "react-native"
 import { AutoImage, Text } from "../../components"
 import { translate } from "../../i18n"
 import { colors, spacing } from "../../theme"
 import { Speaker } from "../../services/api/webflow-api.types"
 import { SocialButtons } from "../../components/SocialButton"
 
+const ASSISTANT_GAP = spacing.medium
+const ASSISTANT_WIDTH = (Dimensions.get("window").width - ASSISTANT_GAP * 3) / 2
```
Comment (morganick) [inline app/screens/TalkDetailsScreen/AssistantsList.tsx:10]: This doesn't account for the padding of the container. Based on the calculation, there is no gap as 3 * 16 is 48. This is the padding of the container making the gap 0. Any reason for us not to use % for width?

**Review by morganick — COMMENTED** [review]
I added screenshots to the main comments. I think we could get away with making the width of the assistants 48% with `space-between` on the container so we account for the container spacing and wouldn't have to update it if that spacing changes.

File: app/screens/TalkDetailsScreen/AssistantsList.tsx (hunk 2 above)
Comment (mazenchami) [reply app/screens/TalkDetailsScreen/AssistantsList.tsx:10]: great point, not sure why i did this in the first place 🤔 updated in next push

File: app/screens/TalkDetailsScreen/AssistantsList.tsx [hunk 3]
```diff
@@ -50,12 +53,11 @@ export function AssistantsList(props: AssistantsListProp) {
 
 const $assistantsContainerWithOne: ViewStyle = {
   flexDirection: "row",
-  marginStart: spacing.large,
 }
 
 const $assistantsContainerWithMore: ViewStyle = {
   flexDirection: "row",
-  justifyContent: "space-around",
+  flexWrap: "wrap",
```
[the remark this answers is not in the document]
Comment (mazenchami) [reply app/screens/TalkDetailsScreen/AssistantsList.tsx:54]: combined in next push

File: app/screens/TalkDetailsScreen/AssistantsList.tsx [hunk 4]
```diff
@@ -17,15 +16,13 @@ export function AssistantsList(props: AssistantsListProp) {
 
   return (
     <>
-      <View style={$assistantContainer}>
+      <View style={$assistantRoot}>
```
Comment (morganick) [inline app/screens/TalkDetailsScreen/AssistantsList.tsx:19]: Last minor detail. If we have a root view, do we need the containing `<></>`?

Comment (mazenchami) [reply app/screens/TalkDetailsScreen/AssistantsList.tsx]: good eye. fixed here: https://github.com/infinitered/ChainReactApp2023/pull/166/commits/f06e85655e44e0efd1f8fd0db04f18eebbca77dd

**Pull request merged**